pkg/workloads/console/runner/runner.go

```diff
@@ -736,10 +769,56 @@ func (c *Runner) GetAttachablePod(csl *workloadsv1alpha1.Console) (*corev1.Pod,
 	}
 
 	for _, c := range pod.Spec.Containers {
-		if c.TTY {
+		if !csl.Spec.Interactive || c.TTY {
 			return pod, c.Name, nil
 		}
 	}
 
 	return nil, "", errors.New("no attachable pod found")
 }
+
+// NoninteractiveAttacher knows how to attach to stdio of an existing container, relaying io
+// to the parent process file descriptors.
+type NoninteractiveAttacher struct {
```

Comment:
Did we create a new type to avoid having to break the public interface with a new parameter on the `Attach` func? Just curious really - while there is quite a bit of duplication I can definitely see an argument for either option.

Reply:
It was a personal taste thing really - I felt that there were getting to be so many `if csl.Spec.Interactive` blocks that it was getting hard to follow the logic in any particular place.

Reply:
Yeah that's fair! It definitely would add a couple more conditional branches. Don't mind keeping it as is

I do wonder whether we should be exposing these structs publicly within the package or not. We're already changing the 'public interface' by removing the `Attacher` struct. Given that external code shouldn't have a reason to create a `NonInteractiveAttacher` etc maybe we should make these types private so as to allow for any future refactoring?

Reply:
I think that's a good idea.